src/common/catalog/src/lib.rs

```diff
@@ -64,6 +64,19 @@ pub fn parse_catalog_and_schema_from_db_string(db: &str) -> (&str, &str) {
     }
 }
 
+/// Attempt to parse catalog and schema from given database name
+///
+/// Similar to [`parse_catalog_and_schema_from_db_string`] but returns an optional
+/// catalog if it's not provided in the database name.
+pub fn parse_optional_catalog_and_schema_from_db_string(db: &str) -> (Option<&str>, &str) {
+    let parts = db.splitn(2, '-').collect::<Vec<&str>>();
+    if parts.len() == 2 {
+        (Some(parts[0]), parts[1])
+    } else {
+        (None, db)
+    }
+}
```

Comment:
we can modify `parse_catalog_and_schema_from_db_string` to call this function to avoid duplicated logic.